code/builders/builder-webpack5/src/index.ts

```diff
@@ -195,8 +196,7 @@ const starter: StarterFunction = async function* starterGeneratorFn({
   }
 
   if (stats.hasErrors()) {
-    // eslint-disable-next-line @typescript-eslint/no-throw-literal
-    throw stats;
+    throw new WebpackCompilationError({ error: stats });
```

Comment:
Don't merge this suggested code, it's just a starting point for the convo:

It seems weird to me that we'd lump these different types of issues together.
Should we separate these into multiple different  
```suggestion
    throw new WebpackCompilationStatsError({ stats });
```

Or we separate the different input types like so:
```suggestion
    throw new WebpackCompilationError({ stats });
```

WDTY?